CMakeLists.txt

```diff
@@ -0,0 +1,115 @@
+# CMake Project for SDL_gpu_shadercross - Simple DirectMedia Layer Shader Cross Compiler
+# Written by @thatcosmonaut
+cmake_minimum_required(VERSION 3.5)
+project(SDL_gpu_shadercross C)
+find_package(SDL3 REQUIRED)
+
+# Options
+option(BUILD_STATIC "Build static library" ON)
+option(BUILD_CLI "Build command line executable" ON)
+
+# Version
+SET(LIB_MAJOR_VERSION "1")
+SET(LIB_MINOR_VERSION "0")
+SET(LIB_REVISION "0")
+SET(LIB_VERSION "${LIB_MAJOR_VERSION}.${LIB_MINOR_VERSION}.${LIB_REVISION}")
+
+# Build Type
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+	# By default, we use Release
+	message(STATUS "Setting build type to 'Release' as none was specified.")
+	set(CMAKE_BUILD_TYPE "Release" CACHE
+		STRING "Choose the type of build." FORCE
+	)
+
+	# Set the possible values of build type for cmake-gui
+	set_property(CACHE CMAKE_BUILD_TYPE PROPERTY
+		STRINGS "Debug" "Release" "RelWithDebInfo"
+	)
+endif()
+
+# Platform Flags
+if(APPLE)
+	set(CMAKE_MACOSX_RPATH ON)
+	set(CMAKE_OSX_DEPLOYMENT_TARGET 11.0)
+	set(LOBJC "objc")
+elseif(WIN32)
+	# "SDL_gpu_shadercross.dll", not "libSDL_gpu_shadercross.dll"
+	set(CMAKE_SHARED_LIBRARY_PREFIX "")
+endif()
+
+# Source lists
+
+file(GLOB SOURCE_FILES
+	# Public Headers
+	SDL_gpu_shadercross.h
+	# Source Files
+    src/SDL_gpu_shadercross.c
+    src/spirv_cross_c.h
+    src/spirv.h
+)
+
+add_library(SDL_gpu_shadercross SHARED ${SOURCE_FILES})
```

Comment:
Let's add a `-shader` suffix to the shared library, like you did to the static variant.
(This is also done in all other CMake project)
(Then you need to set the `OUTPUT_NAME` property to make sure the `-static` does not en up in the filename))
At the end of the file, we can create a suffix-less target alias.